File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-4.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack=[]
        for x in tokens:
            if x not in {"+","-","/","*"}:
                stack.append(int(x))
            else:
                if x=="+":
                    a= stack.pop()
                    b=stack.pop()
                    stack.append(a+b)
                elif x=="*":
                    a= stack.pop()
                    b=stack.pop()
                    stack.append(a*b)
                elif x=="/":
                    a= stack.pop()
                    b=stack.pop()
                    stack.append(int(b/a))
                elif x=="-":
                    a= stack.pop()
                    b=stack.pop()
                    stack.append(b-a)
            
        return stack[0]
            

File: Data Structures & Algorithms/daily-temperatures/submission-4.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        ArrayDeque<Integer>stack = new ArrayDeque<>();
        int[] result = new int[n];
        for(int i=n-1;i>-1;i--){
            while(!stack.isEmpty() && temperatures[stack.peek()]<=temperatures[i]){
                stack.pop();
            }
            if(!stack.isEmpty()){
                result[i]=stack.peek()-i;
            }
            stack.push(i);
        }
        return result;
    }
}
